server/src/main/java/org/apache/druid/segment/realtime/firehose/ChatHandlerResource.java

```diff
@@ -73,6 +74,6 @@ public Object doTaskChat(@PathParam("id") String handlerId, @Context HttpHeaders
       return handler.get();
     }
 
-    throw new BadRequestException(StringUtils.format("Can't find chatHandler for handler[%s]", handlerId));
+    throw new ServiceUnavailableException(StringUtils.format("Can't find chatHandler for handler[%s]", handlerId));
```

Comment:
Deserves a comment: something like "Return HTTP 503 so SpecificTaskRetryPolicy retries in case the handler is not registered yet."